### ISSUE TITLE
[Dart] Transformers handling of Input, Output, HostBinding and HostListener

### ISSUE THREAD

**Issue by vicb** [angular/angular#4407]
As of today the `Input`, `Output`, `HostBinding` and `HostListener` are added to the `propMetadata` property of a `RegisteredType`.

As a result `Property` does not generate setters and `Event` does not generate getters.

I think that a best way to handle them would be:
- to merge the info into the metadata 
- to not record the property decorator on the `RegisteredType`

```typescript
@Component({})
class Cmp {
  @Input() pa;
  @Input('pb-name') pb;

  @Output() ea;
  @Output('EventName') eb; 

  @HostBinding('hpa') toHost;

  @HostListener('event', [exp1, exp2]) handleEvent(a1, a2) { ... }
}

// should lead to the same metadata as:

@Component({
  properties: ['pa', 'pb: pb-name'],
  events: ['ea', 'eb: EventName'],
  host: {
    'hpa': 'toHost',
    '(event)': 'handleEvent(exp1, exp2)'  
  }
})
class Cmp {}
```

/cc @vsavkin

**Comment by kegluneq** [comment]
This approach makes sense to me, however I do not have the cycles to make it happen currently.

In addition, the work done on the [compiler-integration](https://github.com/angular/angular/tree/compiler-integration) branch will obsolete any effort put into this now, so I recommend we wait until that has landed to make these sugar-y kind of changes.

**Comment by yjbanov** [comment]
We should also wait until the `Event` rename is done, and we should postpone the rename till after pre-compilation lands on master.

**Comment by yjbanov** [comment]
This is blocked by #3605

**Comment by tejainece** [comment]
Any updates on this?
